legacy/ui/legacy/src/main/java/com/fsck/k9/ui/messageview/MessageViewFragment.kt

```diff
@@ -512,7 +527,14 @@ class MessageViewFragment :
             return
         }
 
-        startRefileActivity(FolderOperation.MOVE, ACTIVITY_CHOOSE_FOLDER_MOVE)
+        val input = ChooseFolderResultContract.Input(
+            accountUuid = account.uuid,
+            currentFolderId = messageReference.folderId,
+            scrollToFolderId = account.lastSelectedFolderId,
+            messageReference = messageReference,
+        )
+
+        chooseFolderToMoveLauncher.launch(input)
```

Comment:
Optional: Since the `ChooseFolderResultContract.Input` properties are essentially arguments to `launch()`, I think it feels more natural to include the data class instantiation in the `launch()` statement.

```suggestion
        chooseFolderToMoveLauncher.launch(
            ChooseFolderResultContract.Input(
                accountUuid = account.uuid,
                currentFolderId = messageReference.folderId,
                scrollToFolderId = account.lastSelectedFolderId,
                messageReference = messageReference,
            ),
        )
```

Alternatively, we could create an extension function to hide `ChooseFolderResultContract.Input` from the caller:

```kotlin
fun ActivityResultLauncher<ChooseFolderResultContract.Input>.launch(
    accountUuid: String,
    currentFolderId: Long? = null,
    scrollToFolderId: Long? = null,
    messageReference: MessageReference? = null,
) {
    launch(input = ChooseFolderResultContract.Input(accountUuid, currentFolderId, scrollToFolderId, messageReference))
}
```

Reply:
I'm agree to include data class instantiation on  launch, indeed I just moved the shared code on `startRefileActivity` and I didn´t realize that.
For the second suggestion, I thought it's nicer a function signature with only one argument, isn't it? Anyway I'll do what you prefer.